ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/FileOperations.java

```diff
@@ -167,10 +169,25 @@ private void writeFilesList(FileStatus[] fileStatuses, BufferedWriter writer, St
   }
 
   private FileStatus[] listFilesInDir(Path path) throws IOException {
-    return dataFileSystem.listStatus(path, p -> {
-      String name = p.getName();
-      return !name.startsWith("_") && !name.startsWith(".");
-    });
+    boolean done = false;
+    int repeat = 0;
+    FileStatus[] files = null;
+    while (!done) {
+      try {
+        files =  dataFileSystem.listStatus(path, p -> {
+          String name = p.getName();
+          return !name.startsWith("_") && !name.startsWith(".");
+        });
+        done = true;
+      } catch (IOException e) {
+        if (++repeat == MAX_IO_ERROR_RETRY) {
+          throw e;
+        }
+        // in case of io error, reset the file system object
+        dataFileSystem = path.getFileSystem(hiveConf);
+      }
+    }
+    return files;
   }
 
```

Comment:
How about FileOperations.exportFilesAsList? It can also fail with IO error.

Reply:
there are many i/o failure scenario ..i am not sure if we need to handle all as part of this patch

Reply:
move retry to exportFilesAsList method